src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests.vb

```diff
@@ -4452,6 +4452,321 @@ End Class
+        Public Sub Attribute_GenericTypeInParameter_Constructor()
+            Dim source =
+<compilation>
+    <file><![CDATA[
+Class A
+    Inherits System.Attribute
+
+    Public Sub New(e As B(Of Integer).E)
+    End Sub
+End Class
+
+Public Class B(Of T)
+    Public Enum E
+        X
+    End Enum
+
+    Public Const C As E = CType(33, E)
+End Class
+
+<A(B(Of Integer).C)>
+Class C
+End Class
+    ]]></file>
+</compilation>
+
+            Dim verifier = CompileAndVerify(source, symbolValidator:=
+                Sub(m)
+                    Dim c = m.GlobalNamespace.GetTypeMember("C")
+                    Dim attr = c.GetAttributes().Single(Function(d) d.AttributeClass?.Name = "A")
+                    Assert.False(attr.HasErrors)
+                    Assert.Empty(attr.NamedArguments)
+                    Dim arg = attr.ConstructorArguments.Single()
+                    Assert.Equal(33, arg.Value)
+                    Assert.Equal("B(Of System.Int32).E", arg.Type.ToTestDisplayString())
+                End Sub)
+            verifier.VerifyDiagnostics()
+        End Sub
+
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66370")>
+        Public Sub Attribute_GenericTypeInParameter_Property()
+            Dim source =
+<compilation>
+    <file><![CDATA[
+Class A
+    Inherits System.Attribute
+
+    Public Property E As B(Of Integer).E
+End Class
+
+Public Class B(Of T)
+    Public Enum E
+        X
+    End Enum
+
+    Public Const C As E = CType(33, E)
+End Class
+
+<A(E:=B(Of Integer).C)>
+Class C
+End Class
+    ]]></file>
+</compilation>
+
+            Dim verifier = CompileAndVerify(source, symbolValidator:=
+                Sub(m)
+                    Dim c = m.GlobalNamespace.GetTypeMember("C")
+                    Dim attr = c.GetAttributes().Single(Function(d) d.AttributeClass?.Name = "A")
+                    Assert.False(attr.HasErrors)
+                    Assert.Empty(attr.ConstructorArguments)
+                    Dim arg = attr.NamedArguments.Single()
+                    Assert.Equal("E", arg.Key)
+                    Assert.Equal(33, arg.Value.Value)
+                    Assert.Equal("B(Of System.Int32).E", arg.Value.Type.ToTestDisplayString())
+                End Sub)
+            verifier.VerifyDiagnostics()
+        End Sub
+
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66370")>
+        Public Sub Attribute_Generic_Constructor()
+            Dim source1 = <![CDATA[
+class A<T> : System.Attribute
+{
+    public A(T t) { }
+}
+
+struct B<T>
+{
+    public enum E { }
+    public const E C = (E)33;
+}
+
+[A<B<int>.E>(B<int>.C)]
+public class C { }
+]]>
+            Dim comp1 = CreateCSharpCompilation(source1)
+
+            Dim source2 =
+<compilation>
+    <file><![CDATA[
+Class D
+    Inherits C
+End Class
+    ]]></file>
+</compilation>
+            Dim verifier = CompileAndVerify(source2, references:={comp1.EmitToImageReference()}, symbolValidator:=
```

Comment:
>CompileAndVerify [](http://example.com/codeflow?start=27&length=16)

It feels like the usage of `CompileAndVerify` for this test (and other similar tests) is an overkill. It is sufficient to create a compilation and locate symbols in the compilation because symbols are coming from metadata. In fact, the code of this compilation can be empty. We can locate type "C" without using proxy type "D" for that.

Reply:
Yes, that's much better, thanks.